### ISSUE TITLE
Runnin predict without cityscape

### ISSUE THREAD

**Issue by Tetsujinfr** [TobyPDE/FRRN#22]
First, thanks for the new update.
I do not have access to Cityscape, how can I use the pre trained models on other images?
Not sure how to transform pics and how to input them to the net properly.

thanks

Tets

**Comment by TobyPDE** [comment]
Hey, 

You can feed any image into the network as follows:

- Load the image into an RGB numpy tensor
- Rescale the intensity values to [0, 1]
- Make sure that the channel ordering is RGB and not BGR (opencv standard)
- Reshape and dimshuffle the tensor to [1, 3, H, W] where H and W are the image height (rows) and width (cols), respectively. 
- Feed the tensor into the network.

**Comment by rnunziata** [comment]
Would it be possible to post an example of code. For passing a image for prediction.

        val, image =  read standard image RGB.
        predictions, loss = val_fn(??,??)

**Comment by BBarbosa** [comment]
Hi @TobyPDE!
I tried to follow the steps you mentioned here but the predictions only get 2 classes (related with light pink and black colors). 
![segmentation](https://user-images.githubusercontent.com/7063937/29812277-f15766d8-8c9d-11e7-9949-8783c3602b93.PNG)

I also printed the outout prediction matrix and, from what I understood, it should paint the segmented image with another colors.

```
[[[2 2 2 ..., 3 2 3]
  [2 2 2 ..., 3 2 3]
  [2 2 2 ..., 3 3 3]
  ...,
  [9 9 9 ..., 0 0 0]
  [8 9 9 ..., 0 0 0]
  [0 0 9 ..., 0 0 0]]]
```

## Edit 

I made same changes at the  **create_color_label_image** function on **dltools/utility.py** and it solved my problem. 

With the best regards

**Comment by daicoolb** [comment]
@BBarbosa Hi, Can you show me your code here ?

**Comment by daicoolb** [comment]
@TobyPDE I have tried it . but it seems not work

**Comment by BBarbosa** [comment]
Hi @daicoolb!
These are the files I have changed:
- use _mypredict.py_ instead of _predict.py_
- replace the _dltools/utility.py_
- replace _train.py_
- replace _dltools/architectures.py_

I'm using OpenCV 3.3 because it gave me problems trying to load a video file with OpenCV 2.4.
Download files from here [frrn.zip](https://github.com/TobyPDE/FRRN/files/1311326/frrn.zip)

**Comment by hans41** [comment]
@BBarbosa Thanks for your changes!
I have used your new scripts to predict my own data. The "predictions[0]" has now 3 dimensions instead of 2. For example:
(1, 800, 1280) vs (800, 1280)
Why is that?
And with the same input images, if using "mypredict.py" I got different results from what "predict.py" did. It seems the "image" you feed to "pred_fn(image)" is not the same "batch[0]" the author feeds to "val_fn(batch[0], batch[1])".

**Comment by BBarbosa** [comment]
@hans41 you're welcome. I hope it helped in some useful way.
Technically, (1,800,1280) and (800,1280) are equivalent.
With regard to the inputs images, i had problems feeding images to the network with the author's **_provider_** implementation. So, I managed to adapt one for my own following what @TobyPDE  described [here](https://github.com/TobyPDE/FRRN/issues/22#issuecomment-319585210) . The outputs may differ from **_mypredict.py_** to **_predict.py_** once there may be some pre-processing operations skipped.  
This in an example of what I got. It works really good even in an different angle from what it was trained for. 
![ped](https://user-images.githubusercontent.com/7063937/30773878-ae2ad222-a071-11e7-8b1a-a18705270e04.png)

**Comment by hans41** [comment]
@BBarbosa Nice. But my predicted results look quite strange. I used your "myprediction.py" without modifications, as following pre-processing steps: 
-  Read: scipy.misc.imread
-  Resize: cv2.resize
-  Rescale: np.divide(image,255.).astype(np.float32, casting='unsafe')
-  Convert: utility.opencv2tensor(image)
-  Reshape: np.reshape
Is that correct? Thanks for your help!
